### PR TITLE
Support (and request) compressed WebSocket messages

### PR DESCRIPTION
**Pull request by mpetazzoni**

### CODE REVIEW

File: signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java [hunk 1]
```diff
@@ -357,35 +302,149 @@ public void onOpen(Connection connection) {
...
+                log.error("ignoring message with unsupported type {}", type);
+                return;
+            }
 
-                if (dataMessage.getKind() == Kind.DATA) {
-                    LinkedHashMap<String, Object> dataMap = new LinkedHashMap<String, Object>();
-                    dataMap.put("logicalTimestampMs", dataMessage.getLogicalTimestampMs());
-                    dataMap.put("data", dataMessage.getData());
+            // Channel name is the 16 bytes following the binary preamble in the header.
+            String channelName = new String(data, offset + BINARY_PREAMBLE_LENGTH,
```
Comment (tedoc2000) [inline signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java]: add charset here or do we require ascii channel names?

File: signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java [hunk 2]
```diff
@@ -357,35 +302,149 @@ public void onOpen(Connection connection) {
...
             }
         }
 
         @Override
         public void onMessage(String data) {
             try {
-                @SuppressWarnings("unchecked")
-                Map<String, Object> dataMap = objectMapper.readValue(data, Map.class);
+                Map<String, Object> dataMap = objectMapper.readValue(data,
+                        new TypeReference<Map<String, Object>>() {});
```
Comment (tedoc2000) [inline signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java]: No need to new each time. make it a static

File: signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java (hunk 1 above)
Comment (mpetazzoni) [inline signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java]: Yeah it's ASCII.

File: signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java (hunk 2 above)
Comment (mpetazzoni) [inline signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java]: Ack

File: signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java (hunk 1 above)
Comment (tedoc2000) [inline signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java]: better to be explicit then with new String(..., "ASCII")

**Comment by mpetazzoni** [comment]
Addressed feedback, also fixed some Javadoc warnings.

**Pull request merged**